addons/account_voucher/models/account_voucher.py

```diff
@@ -224,19 +240,29 @@ def _convert_amount(self, amount):
 
     @api.multi
     def voucher_pay_now_payment_create(self):
-        payment_methods = self.journal_id.outbound_payment_method_ids
+        if self.voucher_type == 'sale':
+            payment_methods = self.journal_id.inbound_payment_method_ids
+            payment_type = 'inbound'
+            partner_type = 'customer'
+            sequence_code = 'account.payment.customer.invoice'
+        else:
+            payment_methods = self.journal_id.outbound_payment_method_ids
+            payment_type = 'outbound'
+            partner_type = 'supplier'
+            sequence_code = 'account.payment.supplier.invoice'
+        name = self.env['ir.sequence'].with_context(ir_sequence_date=self.date).next_by_code(sequence_code)
         return {
-            'payment_type': 'outbound',
+            'name': name,
```

Comment:
there's no need to give the name, since it will be given at the post() following its value

Reply:
but the payment is not posted to avoid generating move lines twice!

Reply:
you have a point :-)